### PR TITLE
add KMM samples

### PR DESCRIPTION
**Pull request by hjanuschka**
sample on how we use a shared fastfile across multiple apps with and without app-extensions

### CODE REVIEW

**Comment by hjanuschka** [comment]
had to remove app-id's and git urls from internal servers/apps - sorry

**Comment by hjanuschka** [comment]
added our android lanes to the sample 
note: the android lanes need the adb/adb_device/gradle(modified) actions -> see this PR: https://github.com/fastlane/fastlane/pull/1130

File: KroneMultimedia/SharedFastfile.rb [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+  sh "killall iOS Simulator &> /dev/null || true"
+end
+
+desc "Run Unit/UI Tests"
+lane :tests do
+  certificates
+
+  kill_simulator
+  snapshot( {
+      workspace: $self_workspace,
```
Comment (KrauseFx) [inline KroneMultimedia/SharedFastfile.rb]: Seems like the tabs are wrong here. You can use `rubocop -a` and point to this file to fix it :+1:

File: KroneMultimedia/SharedFastfile.rb [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+
+ end
+
+end
+
+lane :build_ipa do
+   before_build
+   certificates
+
+  gym({
```
Comment (KrauseFx) [inline KroneMultimedia/SharedFastfile.rb]: Could you remove all the `{` in the Ruby configuration files and just have `( ... )`?

File: KroneMultimedia/SharedFastfile.rb [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+lane :submit_build do |options| 
+  require 'spaceship'
+  Spaceship::Tunes.login
+  app = Spaceship::Tunes::Application.find($self_idfs[0])
+  
+
+  version = app.edit_version
+
+  
+    #get latest build
```
Comment (KrauseFx) [inline KroneMultimedia/SharedFastfile.rb]: Wrong spacing here too, should be fixed with `rubocop -a` :rocket:

File: KroneMultimedia/SharedFastfile.rb [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+    submission.content_rights_has_rights = true
+    
+
+   submission.export_compliance_available_on_french_store = false
+   submission.export_compliance_contains_proprietary_cryptography = false
+   submission.export_compliance_contains_third_party_cryptography = false
+   submission.export_compliance_is_exempt = false
+   submission.export_compliance_uses_encryption = false
+   submission.export_compliance_app_type = nil
+   submission.export_compliance_encryption_updated = false
```
Comment (KrauseFx) [inline KroneMultimedia/SharedFastfile.rb]: This should actually be part of `deliver`, but I guess you had troubles using `deliver` to submit your app for Review? Is this lane just to submit an existing build for Review, right? 
What do you think about adding this functionality to `deliver`?

**Comment by KrauseFx** [comment]
Hey @hjanuschka, thank you so much for this PR. It looks great, however there are some style improvements before it's ready to be merged. See the comments on how you can fix those automatically.

**Comment by hjanuschka** [comment]
ok i rebased to a single commit, did a quick rubocop, fixed almost all issues (only longlines, global vars, are left)

about the commet that `:submit_build` lane should be inside deliver. i totally agree you that this should be in deliver, maybe i am doing a PR there too.

we usually have the workflow:
- create app version on iTC
- invite testflight testers
- submit a new TF build on every git commit - auto increment build nr.
- invite external testers
- resign ipa and invite crowdtesting platforms (won't do this on every app/version)
- once we get a "clear" sign from the internal QA (wich coordinates the testers)
- we may have even newer builds already deployed to testflight, but the "clear" build should go into release, therefore we needed the submit_build
- once the build is submitted and the review is done we start again with a new version train.

**Comment by hjanuschka** [comment]
update: did the PR on `deliver` -> https://github.com/fastlane/deliver/pull/548

**Comment by mfurtak** [comment]
@KrauseFx I don't think I am in touch with how you would like examples to be, so I will let you see this through!

**Comment by KrauseFx** [comment]
Hey @hjanuschka, thanks for the pull request, however we're working on the mono repo right now and won't merge this before the switch. Let's re-connect about this in a few weeks :+1:

**Comment by hjanuschka** [comment]
@KrauseFx  i would love to reopen that one - until now we even added `sharethemeal` and `aws` testing - wich i would provide with a further PR - whats your opinion on that?

we are using it for following apps:
- krone news (ios & android)
- krone sport (ios & android)
- willi (ios & android, www.willi.at)

and a few inhouse apps, with the policy -> no app without shared fastlane!

**Pull request closed without merging**